scheme/ischeme.go

```diff
@@ -10,16 +10,59 @@ import "github.com/veraison/services/proto"
 // to a particular AttestationFormat, such as knowledge of evidence and
 // endorsements structure.
 type IScheme interface {
+	// GetName returns the Scheme name as a string. The name is
+	// typically taken from the corresponding attestation format (it is,
+	// however, theoretically possible for them to be different, e.g. if
+	// there are multiple schemes associated with the same format) .
 	GetName() string
+
+	// GetFormat returns the AttestationFormat associated with this scheme.
 	GetFormat() proto.AttestationFormat
+
+	// GetSupportedMediaTypes returns a string slice of MIME media types
+	// handled by this scheme.
 	GetSupportedMediaTypes() []string
 
-	ExtractVerifiedClaims(token *proto.AttestationToken, trustAnchor string) (*ExtractedClaims, error)
+	// GetTrustAnchorID returns a string ID used to retrieve a trust anchor
+	// for this token. The trust anchor may be necessary to validate the
+	// token and/or extract its claims (if it is encrypted).
 	GetTrustAnchorID(token *proto.AttestationToken) (string, error)
-	AppraiseEvidence(ec *proto.EvidenceContext, endorsements []string) (*proto.AppraisalContext, error)
 
-	// endorsement lookup keys
+	// ExtractClaims parses the attestation token and returns claims
+	// extracted therefrom.
+	ExtractClaims(
+		token *proto.AttestationToken,
+		trustAnchor string,
+	) (*ExtractedClaims, error)
+
+	// ValidateToken verifies the structural integrity and validity of the
+	// token. The exact checks performed are scheme-specific, but they
+	// would typically involve, at the least, verifying the token's
+	// signature using the provided trust anchor. If the validation fails,
+	// an error detailing what went wrong is returned.
+	// TODO(setrofim): no distinction is currently made between validation
+	// failing due to an internal error, and it failing due to bad input
+	// (i.e. signature not matching).
+	ValidateToken(
```

Comment:
I think `ValidateEvidence` maybe a bit too general, as this is specifically intended to validate things like the signature, and other "structural things", rather than the evidence claims. Maybe, `ValidateEvidenceIntegrity`?